hal/src/nRF52840/usb_hal_cdc.c

```diff
@@ -239,37 +279,38 @@ static void usbd_user_ev_handler(app_usbd_event_type_t event)
         }
         case APP_USBD_EVT_DRV_RESUME: {
             LOG_DEBUG(TRACE, "APP_USBD_EVT_DRV_RESUME");
-            set_usb_state(HAL_USB_STATE_DEFAULT);
+            set_usb_state(nrf_usb_state_to_hal_usb_state(app_usbd_core_state_get()));
```

Comment:
Since we have corresponding Nordic USB state, do we need to set the state again? What about wrapping Nordic API inside `HAL_USB_State HAL_USB_Get_State()`
```
HAL_USB_State HAL_USB_Get_State() {
    return nrf_usb_state_to_hal_usb_state(app_usbd_core_state_get();
}
```

Reply:
Well, we need to synchronize it, because there is no corresponding `SUSPENDED` state in Nordic API.

This `HAL_USB_Get_State()` also wouldn't work, because we still need to execute the callback to notify about the state transitions.